functions/_fzf_preview_changed_file.fish

```diff
@@ -0,0 +1,20 @@
+function _fzf_preview_changed_file
+    set -l path (string split ' ' $argv)[-1]
+    if string match -r '^\?\?' $argv --quiet
```

Comment:
Hi, I'm back! I've found a great way to extract the symbols without using regex--you can use `git status --porcelain`. Can you update the code to use that instead? Afterwards, we'll be very close to merging

Reply:
I didn't quite understand it and didn't find its use effect

![image](https://user-images.githubusercontent.com/12210746/172096898-56828210-8c71-4c58-86d4-23316742486a.png)

Reply:
Try using version 1 porcelain and qualifying the file path. The status it displays is much easier to parse because it's at the very beginning of the line. Then you can use substring instead of playing with unreadable regexes.

Reply:
> 

but it still needs to regex with prefix to detect file status

Reply:
What I'd strongly prefer to see is something like this pattern
```fish
set file_status (command to grab the two letters of the file status)
if test file_status[0] = "A"

```
I think this'll be much more readable. Right now I still don't exactly know what your regexes are matching.

Reply:
> What I'd strongly prefer to see is something like this pattern
> 
> ```fish
> set file_status (command to grab the two letters of the file status)
> if test file_status[0] = "A"
> ```
> 
> I think this'll be much more readable. Right now I still don't exactly know what your regexes are matching.

`--porcelain` no diff to `-s`
![image](https://user-images.githubusercontent.com/12210746/173193794-f14c3e09-e22b-4dfb-9831-5802ed9a1c5e.png)

Reply:
> What I'd strongly prefer to see is something like this pattern
> 
> ```fish
> set file_status (command to grab the two letters of the file status)
> if test file_status[0] = "A"
> ```
> 
> I think this'll be much more readable. Right now I still don't exactly know what your regexes are matching.

`git status` shows with two character, when first not null, it means staged changes, while second, the unstaged.

So, `\S\s\S` means unstaged and `^\S` means staged

Reply:
Got it, I get it now, thanks! I am fine with the logic as it is then.